### PR TITLE
Add Evaluation Function

### PR DESCRIPTION
**Pull request by ysnows**
Evaluating is an important part of the LLM development process, so I added the Evaluation function to LangchaiJS, which includes the following features:

1. String Evaluators
    a . Criteria Evaluation
    b. Embedding Distance Evaluation
2. Comparison Evaluators
    a. Pairwise String Comparison
    b. Pairwise Embedding Distance
3. Trajectory Evaluators
    a. Agent Trajectory

All implementations have referred to the Python version of [Langchain](https://python.langchain.com/docs/guides/evaluation/)

### CODE REVIEW

**Comment by jacoblee93** [comment]
You rock! Will look today. CC @hinthornw

**Comment by jacoblee93** [comment]
Thank you so much for this!

File: langchain/src/evaluation/agents/prompt.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+[RESPONSE]
+{answer}
+[END_RESPONSE]
+
+Let's do a detailed evaluation of the AI language model's answer step by step.
+
+We consider the following criteria before giving a score from 1 to 5:
+
+i. Is the final answer helpful?
+    ii. Does the AI language use a logical sequence of tools to answer the question?
```
Comment (hinthornw) [inline langchain/src/evaluation/agents/prompt.ts:39]: Maybe this is just git rendering, but looks like there're some unwanted indentations here?

File: langchain/src/evaluation/agents/prompt.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+[RESPONSE]
+The AI language model's final answer to the question was: There are different ways to measure the length of the United States, but if we use the distance between the Statue of Liberty and the westernmost point of the contiguous United States (Cape Alava, Washington), which is approximately 2,857 miles (4,596 km), and assume that the Statue of Liberty is 305 feet (93 meters) tall, then the statue would stretch across the United States approximately 17.5 times if laid end to end.
+[END_RESPONSE]
+
+Let's do a detailed evaluation of the AI language model's answer step by step.
+
+We consider the following criteria before giving a score from 1 to 5:
+
+i. Is the final answer helpful?
+    ii. Does the AI language use a logical sequence of tools to answer the question?
```
Comment (hinthornw) [inline langchain/src/evaluation/agents/prompt.ts:84]: Same here

**Review by hinthornw — APPROVED** [review]

**Comment by ysnows** [comment]
> Thank you so much for this!

> You're welcome, and  thank you for your review and refactor. 😀 @jacoblee93 @hinthornw

**Comment by jacoblee93** [comment]
Legend

**Pull request merged**